tracer/src/Datadog.Trace.Bundle/Datadog.Trace.Bundle.targets

```diff
@@ -0,0 +1,77 @@
+      <EffectiveTargetFramework Condition="'$(TargetFramework)' == 'net8.0'">net6.0</EffectiveTargetFramework>
+      <EffectiveTargetFramework Condition="'$(TargetFramework)' == 'net9.0'">net6.0</EffectiveTargetFramework>
+      <EffectiveTargetFramework Condition="'$(TargetFramework)' == 'net10.0'">net6.0</EffectiveTargetFramework>
+    </PropertyGroup>
+
+    <Message Importance="High" Text="[Datadog.Trace.Bundle] TargetFramework: $(TargetFramework) => $(EffectiveTargetFramework)" />
+    <Message Importance="High" Text="[Datadog.Trace.Bundle] RuntimeIdentifier: $(RuntimeIdentifier), Platform: $(Platform)" Condition="'$(RuntimeIdentifier)' != ''" />
+    <Message Importance="High" Text="[Datadog.Trace.Bundle] RuntimeIdentifier: (not set), Platform: $(Platform)" Condition="'$(RuntimeIdentifier)' == ''" />
+
+    <ItemGroup>
+      <AllLibDirs Include="$([System.IO.Directory]::GetDirectories('$(BundleRoot)'))" />
+
+      <AllManagedLibDirs Include="$([System.IO.Directory]::GetDirectories('$(BundleRoot)', 'net*'))" />
+
+      <AllNativeLibDirs Include="$([System.IO.Directory]::GetDirectories('$(BundleRoot)', 'win-*'))" />
+      <AllNativeLibDirs Include="$([System.IO.Directory]::GetDirectories('$(BundleRoot)', 'linux-*'))" />
+      <AllNativeLibDirs Include="$([System.IO.Directory]::GetDirectories('$(BundleRoot)', 'osx*'))" />
+
+      <!-- if .NET Framework
+             if Platform == x64, keep "win-x64"
+             if Platform == x86, keep "win-x86"
+             else keep "win-*" as fallback
+       -->
+      <KeepNativeLibDirs Include="$(BundleRoot)win-x64"
+                         Condition="$(IsNetFramework) AND '$(Platform)' == 'x64'" />
+      <KeepNativeLibDirs Include="$(BundleRoot)win-x86"
+                         Condition="$(IsNetFramework) AND '$(Platform)' == 'x86'" />
+      <KeepNativeLibDirs Include="$([System.IO.Directory]::GetDirectories('$(BundleRoot)', 'win-*'))"
+                         Condition="$(IsNetFramework) AND '$(Platform)' != 'x64' AND'$(Platform)' != 'x86'"/>
+
+      <!-- if .NET Core
+             if $(RuntimeIdentifier) is set, keep directory "$(RuntimeIdentifier)"
+             else keep "win-*;linux-*;osx-*" as fallback -->
+      <KeepNativeLibDirs Include="$(BundleRoot)$(RuntimeIdentifier)"
+                         Condition="!$(IsNetFramework) AND '$(RuntimeIdentifier)' != ''"/>
+      <KeepNativeLibDirs Include="@(AllNativeLibDirs)"
+                         Condition="!$(IsNetFramework) AND '$(RuntimeIdentifier)' == ''"/>
+
+      <!-- if $(EffectiveTargetFramework) is set, keep directory "$(EffectiveTargetFramework)"
+           else keep "net*" as fallback -->
+      <KeepManagedLibDirs Include="$(BundleRoot)$(EffectiveTargetFramework)"
+                          Condition="'$(EffectiveTargetFramework)' != ''"/>
+      <KeepManagedLibDirs Include="$([System.IO.Directory]::GetDirectories('$(BundleRoot)', 'net*'))"
```

Comment:
What are the scenarios where `'$(EffectiveTargetFramework)' == ''`? I'm assuming that there will always be a `$(TargetFramework)` for running the publish operation, so does this only happen when the specified `$(TargetFramework)` is lower than our supported versions?

Reply:
I added an error message if `EffectiveTargetFramework` empty, which means we don't support the `TargetFramework`.